src/gluonts/evaluation/_base.py

```diff
@@ -259,15 +269,15 @@ def get_metrics_per_ts(
         self, time_series: Union[pd.Series, pd.DataFrame], forecast: Forecast
     ) -> Dict[str, Union[float, str, None]]:
         pred_target = np.array(self.extract_pred_target(time_series, forecast))
-        pred_target = np.ma.masked_invalid(pred_target)
-
-        # required for seasonal_error and owa calculation
         past_data = np.array(self.extract_past_data(time_series, forecast))
-        past_data = np.ma.masked_invalid(past_data)
+
+        if self.mask_invalid_timeseries:
+            past_data = np.ma.masked_invalid(past_data)
+            pred_target = np.ma.masked_invalid(pred_target)
 
         try:
             mean_fcst = forecast.mean
```

Comment:
I think this can just be `mean_fcst = getattr(forecast, "mean", None)`, so we would get around the `try-except` entirely.

Reply:
Nice!

Reply:
This breaks the tests, as some forecasts have the attribute but raises a `NotImplementedError`.